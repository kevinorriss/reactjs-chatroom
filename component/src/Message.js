import React from 'react'

class Message extends React.Component {
    render() {
        const { username, text, isNotification, isMe } = this.props
        if (isNotification) {
            return (
                <p className="chatroom__notification">{text}</p>
            )
        } else {
            return (
                <div className={"chatroom__message" + (isMe ? " chatroom__message--me" : "")}>
                    <h5>{username}</h5>
                    <p>{text}</p>
                </div>
            )
        }
    }
}

export default Message