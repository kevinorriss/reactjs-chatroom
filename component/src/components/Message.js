import React from 'react'
import PropTypes from 'prop-types'

class Message extends React.Component {
    render() {
        const { username, text, isNotification, isError, isMe } = this.props
        if (isError) {
            return (
                <p className="chatroom__notification chatroom__notification--error">{text}</p>
            )
        } else if (isNotification) {
            return (
                <p className="chatroom__notification">{text}</p>
            )
        } else {
            return (
                <div className={"chatroom__message" + (isMe ? " chatroom__message--me" : "")}>
                    <h5>{username}</h5>
                    <p>{text}</p>
                </div>
            )
        }
    }
}

Message.propTypes = {
    username: PropTypes.string,
    text: PropTypes.string.isRequired,
    isNotification: PropTypes.bool,
    isError: PropTypes.bool,
    isMe: PropTypes.bool
}

export default Message