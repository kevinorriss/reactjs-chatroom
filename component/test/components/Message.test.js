import Message from '../../src/components/Message'

// disable console errors for cleaner test output
jest.spyOn(console, 'error').mockImplementation(() => {})

test('Should render received message correctly', () => {
    expect(shallow( <Message username='Test User' text='This is a received message' /> )).toMatchSnapshot()
})

test('Should render send message correctly', () => {
    expect(shallow( <Message username='Myself' text='This is a sent message' isMe={true} /> )).toMatchSnapshot()
})

test('Should render notification message correctly', () => {
    expect(shallow( <Message username='Myself' text='This is a notification' isNotification={true}/> )).toMatchSnapshot()
})

test('Should render error message correctly', () => {
    expect(shallow( <Message username='Myself' text='This is an error' isError={true}/> )).toMatchSnapshot()
})

test('Should render error when message is both notification and error', () => {
    expect(shallow( <Message username='Test User' text='This is a notification and an error' isError={true} isNotification={true}/> )).toMatchSnapshot()
})