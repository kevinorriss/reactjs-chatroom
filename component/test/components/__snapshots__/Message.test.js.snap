// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render error message correctly 1`] = `
<p
  className="chatroom__notification chatroom__notification--error"
>
  This is an error
</p>
`;

exports[`Should render error when message is both notification and error 1`] = `
<p
  className="chatroom__notification chatroom__notification--error"
>
  This is a notification and an error
</p>
`;

exports[`Should render notification message correctly 1`] = `
<p
  className="chatroom__notification"
>
  This is a notification
</p>
`;

exports[`Should render received message correctly 1`] = `
<div
  className="chatroom__message"
>
  <h5>
    Test User
  </h5>
  <p>
    This is a received message
  </p>
</div>
`;

exports[`Should render send message correctly 1`] = `
<div
  className="chatroom__message chatroom__message--me"
>
  <h5>
    Myself
  </h5>
  <p>
    This is a sent message
  </p>
</div>
`;
